### PR TITLE
12:feat/normalize-feature-files

### LINKED ISSUE #12: Preprocessing: Implement function that makes the length of all video features the same
Though the number of keypoints/landmarks in each frame is the same, each video can have a different number of frames, so if a feature array generated from a video has dimensions $n \times d$, then $d$ will be the same for all videos (each frame has the same number of keypoints), but $n$, representing the number of frames, will not be the same, which will cause issues for sklearn SVM training. 

One way to implement this would be to find the video with the largest number of frames, let that be $m$, then add frames with all 0s to every other video's feature array so that they all end up having a total of $m$ frames, causing every video's feature array to be $m \times d$ 2D arrays. If you can think of a better implementation, then feel free to do it your own way instead.

This is definitely something that has to be implemented for SVM and likely k-nearest neighbour training, but I'm not sure if its needed for neural networks.

### PR DESCRIPTION
**Pull request by HaNguyenCS**
resolve #12 

Moreover, codes added to skip files that are
- empty (this has to be addressed, in issue 15)
- named ordered_labels.npy because this file is only 1D

### CODE REVIEW

**Review by GauravS43 — APPROVED** [review]

**Review by PourquoiEstTu — CHANGES_REQUESTED** [review]

**Review by GauravS43 — APPROVED** [review]

**Review by PourquoiEstTu — APPROVED** [review]
Looks good, anyone running this file still may have to change the `BASE_DIR` variable depending on where the dataset is located for them, but its not a problem.

File: backend/app/preprocess.py [hunk 1]
```diff
@@ -197,6 +207,57 @@ def get_labels_sklearn(features_dir:str, json_path: str=JSON_PATH, overwrite_pre
...
+       input_dir: directory with features generated from gen_videos_features()
+       output_dir: directory where processed files are saved"""
+
+    max_length = 0
+    for file in os.scandir(input_dir) :
+        if file.is_file() and file.name.endswith(".npy"):
+            arr = np.load(file.path)
+            n_frames = arr.shape[0] # number of rows/frames
+            max_length = max(max_length, n_frames)
+    print(f"[normalize_sequence_length] Max frame length found: {max_length}")
```
Comment (PourquoiEstTu) [inline backend/app/preprocess.py]: The max length code takes into account the `ordered_labels.npy` file, causing all the normalized files to be roughly 4 megabytes in size when they could be much smaller at around 90 kilobytes. Code to exclude `ordered_labels.npy` from this loop should be added as well. Good spot from @GauravS43.
